sharding/utils/marshal_test.go

```diff
@@ -0,0 +1,80 @@
+package utils
+
+import (
+	"math/rand"
+	"reflect"
+	"testing"
+)
+
+func buildtxblobs(size int64) []interface{} {
+	tempbody := make([]interface{}, size)
+	for i := int64(0); i < size; i++ {
+		tempbody[i] = buildblob(size)
+
+	}
+	return tempbody
+}
+
+func buildblob(size int64) []interface{} {
+
+	tempbody := make([]interface{}, size)
+	for i := int64(0); i < size; i++ {
+		tempbody[i] = byte(rand.Int())
+
+	}
+
+	return tempbody
+
+}
+func TestConvertInterface(t *testing.T) {
+	slice := []interface{}{0, 1, 2, 3, 4, 5}
+	convertedValue, err := ConvertInterface(slice, reflect.Slice)
+	if err != nil {
+		t.Fatalf("Error: %v %v", err, convertedValue)
+	}
+
+}
+func TestSize(t *testing.T) {
+	size := int64(800)
+	blob := buildtxblobs(size)
+	chunksafterSerialize := size / chunkDataSize
+	terminalchunk := size % chunkDataSize
+	if terminalchunk != 0 {
+		chunksafterSerialize = chunksafterSerialize + 1
+	}
+	chunksafterSerialize = chunksafterSerialize * size
+	sizeafterSerialize := chunksafterSerialize * chunkSize
+	serializedblob, err := Serialize(blob)
+	if err != nil {
+		t.Fatalf("Error Serializing blob:%v %v", err, serializedblob)
```

Comment:
Please only call fatal when the test can no longer possibly proceed. Use t.Errorf here and below